osu.Game/Overlays/Volume/VolumeMeter.cs

```diff
@@ -217,6 +227,29 @@ protected double DisplayVolume
 
                 volumeCircle.Current.Value = displayVolume * 0.75f;
                 volumeCircleGlow.Current.Value = displayVolume * 0.75f;
+
+                Schedule(() =>
+                {
+                    const int sfx_debounce_time = 30;
+
+                    if (firstUpdate ||
+                        !displayVolumeChanged ||
+                        Time.Current - sampleLastPlaybackTime <= sfx_debounce_time)
+                    {
+                        firstUpdate = false;
+                        return;
+                    }
```

Comment:
Dunno if it's just me but I find combining the `firstUpdate` logic with general "should play this sample" logic a little overcomplicated. I'd probably prefer writing two separate guards:

```csharp
if (firstUpdate)
{
    firstUpdate = false;
    return;
}

if (!displayVolumeChanged || Time.Current - sampleLastPlaybackTime <= sfx_debounce_time)
{
    return;
}
```

I am guessing a bit as to what the intent here was, so this may not be right if I misread.

Reply:
Yeah, that was the intent.

I originally wrote it as two separate guards, but combined them to avoid the double returns. If you think it's more readable with them separated, then I have no issue doing that.

Reply:
I'm rewriting this function as it's all a bit weird.

Reply:
I've rewritten this function. Probably worth someone else double-checking the new logic. Hopefully reads better.

Reply:
Or I haven't because changes were pushed ahead of mine..

Reply:
Ah I didn't notice the message before pushing, wasn't expecting you to jump in and rewrite 🤷